### DISCUSSION TITLE
Editor (VS Code) integration?

### DISCUSSION THREAD

**Discussion by vfonic** [code-hike/codehike#88] [Ideas and feedback]
This tool looks great!

I'm always looking for ways to improve what I can see in my editor without the need to have to open multiple split panes or some other acrobatics. Having a group of files open one below the other (like in your demo) could be an awesome advantage!

I see many use cases for this:
React: JSX file + GraphQL query in another file, maybe CSS module file or CSS-in-JS,...
Web: HTML, CSS and JS files all open one below the other...

Deep linking: my colleague might ask me where something is happening and I can send her a deep link, kinda like I do now to GitHub code, but why GitHub? Wouldn't it be more practical to see this directly in the editor?

Similarly, code annotations could be built into the editor!

Sorry for the messy message, I'm writing this from my phone, but I just wanted to get it out to you. :)

**Comment by pomber** [comment]
Not sure I understand. You already can open one file below the other on VS Code.

**Reply by vfonic** [reply]
That's true. I'm more thinking in the direction of better collaboration:
- I could share a link to a code and have the editor focus/annotate/highlight/open files with certain layout...

Check the last example here: https://codehike.org/demo/code
Imagine a colleague of yours wants to know what `sit` does. They could send you a link to gh repo, file and line (no column highlighing) and ask: what does this do? or what does `sit` here mean?

Basically writing `sit` as part of the question is kind-of like highlighting it via a specifically crafted link (like you can do for line on github).

I'm thinking out loud here.
I see this being useful for:
1) when you need to point someone to some specific code in the codebase
2) when you want to have the code documentation embedded in the codebase

**Reply by pomber** [reply]
Now I understand. Interesting idea. 

There's already some syntax for "smart links" in Code Hike (https://codehike.org/demo/sections):
```md
Lorem dolor sit amet, [javascript](focus://index.js#2:3) adipiscing elit.
```
which I want to extend to support different types of annotations.

Using this for an editable file inside VS Code would be a lot of work. But maybe it could be used as an alternative to github links, with extra features.
